### PR TITLE
Add deployment workflow template

### PR DESCRIPTION
**Pull request by wibisana-kargo**

### CODE REVIEW

File: .github/workflows/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      deployer_installation_id:
+        required: true
+
+      gke_sa_key:
+        required: true
+
+jobs:
+  deployment:
+    name: Deploy
+    runs-on: ubuntu-20.04
```
Comment (fersandi-kargo) [inline .github/workflows/deployment.yaml]:
```suggestion
    runs-on: self-hosted
```

Or parameterized

File: .github/workflows/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v1
+        with:
+          app_id: ${{ secrets.deployer_app_id }}
+          private_key: ${{ secrets.deployer_private_key }}
+          installation_id: ${{ secrets.deployer_installation_id }}
+          repository: ${{ inputs.repository }}
+
+      - name: Checkout code
```
Comment (fersandi-kargo) [inline .github/workflows/deployment.yaml:80]: Why we need to checkout repo code?

File: .github/workflows/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          token: ${{ steps.generate_token.outputs.token }}
+          path: rhodes
+
+      - name: Set git config
+        run: |
+          AUTHOR_NAME=$(git show -s --format='%aN' ${{ github.sha }})
+          AUTHOR_EMAIL=$(git show -s --format='%ae' ${{ github.sha }})
+          git config --global user.name "${AUTHOR_NAME}"
+          git config --global user.email "${AUTHOR_EMAIL}"
+      - name: Update service tag in rhodes
```
Comment (fersandi-kargo) [inline .github/workflows/deployment.yaml]:
```suggestion
      - name: Update service tag
```

File: .github/workflows/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          export_default_credentials: true  
+
+      - name: Setup Kubernetes credential
+        run: gcloud container clusters get-credentials "${{ inputs.gke_cluster }}" --zone "${{ inputs.gke_zone }}"
+      - name: Change kubernetes namespace to `argocd`
+        run: kubectl config set-context --current --namespace=argocd
+      - name: ArgoCD sync app
+        uses: clowdhaus/argo-cd-action/@main
+        with:
+          command: app sync ${{ inputs.environment }}-backend-${{ inputs.service_name }}
```
Comment (fersandi-kargo) [inline .github/workflows/deployment.yaml]: Hard coded

File: .github/workflows/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+          options: --core --async
+
+      - name: ArgoCD wait until application healthy
+        uses: clowdhaus/argo-cd-action/@main
+        with:
+          command: app wait ${{ inputs.environment }}-backend-${{ inputs.service_name }}
```
Comment (fersandi-kargo) [inline .github/workflows/deployment.yaml]: Hard coded

File: .github/workflows/deployment.yaml (hunk 2 above)
Comment (wibisana-kargo) [reply .github/workflows/deployment.yaml:80]: to get Author information. I try to make commit in target repo similar to source commit

**Review by lantang-kargo — APPROVED** [review]

**Review by ariz-kargo — APPROVED** [review]

**Pull request merged**